cirq-google/cirq_google/calibration/workflow.py

```diff
@@ -649,36 +707,34 @@ def run_calibrations(
 
     if engine is not None:
         if processor_id is None:
-            raise ValueError('processor_id must be provided when running on the engine')
+            raise ValueError('processor_id must be provided when using an Engine')
         if gate_set is None:
-            raise ValueError('gate_set must be provided when running on the engine')
-
-        results = []
+            raise ValueError('gate_set must be provided when using an Engine')
+
+        if calibration_request_type == LocalXEBPhasedFSimCalibrationRequest:
+            sampler = engine.sampler(processor_id=processor_id, gate_set=gate_set)
+            return _run_local_calibrations_via_sampler(calibration_requests, sampler)
+
+        return _run_calibrations_via_engine(
+            calibration_requests,
+            engine,
+            processor_id,
+            gate_set,
+            max_layers_per_request,
+            progress_func,
+        )
 
-        requests = [
-            [
-                calibration.to_calibration_layer()
-                for calibration in calibrations[offset : offset + max_layers_per_request]
-            ]
-            for offset in range(0, len(calibrations), max_layers_per_request)
-        ]
+    if calibration_request_type == LocalXEBPhasedFSimCalibrationRequest:
+        return _run_local_calibrations_via_sampler(
+            calibration_requests, sampler=cast(Sampler, sampler)
+        )
 
-        for request in requests:
-            job = engine.run_calibration(request, processor_id=processor_id, gate_set=gate_set)
-            request_results = job.calibration_results()
-            results += [
-                calibration.parse_result(result)
-                for calibration, result in zip(calibrations, request_results)
-            ]
-            if progress_func:
-                progress_func(len(results), len(calibrations))
-
-    elif isinstance(sampler, PhasedFSimEngineSimulator):
-        results = sampler.get_calibrations(calibrations)
-    else:
-        raise ValueError(f'Unsupported sampler type {type(sampler)}')
+    if isinstance(sampler, PhasedFSimEngineSimulator):
+        return sampler.get_calibrations(calibration_requests)
 
-    return results
+    raise ValueError(
+        f'Unsupported engine/request combinations: {engine} cannot run {calibration_requests}'
```

Comment:
When engine is None, this will read "None cannot run [XEBPhasedFSimCalibrationRequest(...), XEBPhasedFSimCalibrationRequest(...), ...]", also the user passes in `sampler`, not `engine`. I would rather use 

```suggestion
        f'Unsupported sampler/request combinations. Sampler={sampler} cannot run calibration request of type {calibration_request_type}'
```

Reply:
good catch and suggestion. The error was from a merge issue following https://github.com/quantumlib/Cirq/pull/4054